packages/common/src/hardhat/tasks/xchainSetup.ts

```diff
@@ -0,0 +1,206 @@
+      console.log(`...txn: ${setOracleSpokeTxn.transactionHash}`);
+    }
+    if (messengerGovernorSpoke !== GovernorSpoke.address) {
+      console.log(`Setting governor spoke to ${GovernorSpoke.address}...`);
+      const setGovernorSpokeTxn = await messenger.methods
+        .setGovernorSpoke(GovernorSpoke.address)
+        .send({ from: deployer });
+      console.log(`...txn: ${setGovernorSpokeTxn.transactionHash}`);
+    }
+
+    // Submit parent messenger cross-chain transactions:
+    // First, send ETH to the parent messenger to cover both transactions.
+    let messengerBalance = await web3.eth.getBalance(messenger.options.address);
+    if (toBN(messengerBalance.toString()).lt(requiredEthForOneTransaction)) {
+      console.log(`Sending ${requiredEthForOneTransaction.toString()} ETH to the messenger`);
+      const sendEthTxn = await web3.eth.sendTransaction({
+        to: messenger.options.address,
+        from: deployer,
+        value: requiredEthForOneTransaction.toString(),
+      });
+      console.log(`...txn: ${sendEthTxn.transactionHash}`);
+    }
+    console.log(`Setting child oracle spoke address to ${OracleSpoke.address}...`);
+    const setChildOracleSpokeTxn = await messenger.methods
+      .setChildOracleSpoke(OracleSpoke.address)
+      .send({ from: deployer });
+    console.log(`...txn: ${setChildOracleSpokeTxn.transactionHash}`);
+    messengerBalance = await web3.eth.getBalance(messenger.options.address);
+    if (toBN(messengerBalance.toString()).lt(requiredEthForOneTransaction)) {
+      console.log(`Sending ${requiredEthForOneTransaction.toString()} ETH to the messenger`);
+      const sendEthTxn = await web3.eth.sendTransaction({
+        to: messenger.options.address,
+        from: deployer,
+        value: requiredEthForOneTransaction.toString(),
+      });
+      console.log(`...txn: ${sendEthTxn.transactionHash}`);
+    }
+    console.log(`Setting child parent messenger to ${messenger.options.address}...`);
+    const setChildParentMessengerTxn = await messenger.methods
+      .setChildParentMessenger(messenger.options.address)
+      .send({ from: deployer });
+    console.log(`...txn: ${setChildParentMessengerTxn.transactionHash}`);
+
+    // Submit oracle hub transactions:
+    await setupHub(oracleHub, deployer, messenger.options.address, 42161);
+
+    // Submit governor hub transactions:
+    await setupHub(governorHub, deployer, messenger.options.address, 42161);
+  }
+);
+
+task("setup-l2-arbitrum-xchain", "Configures L2 cross chain smart contracts for Arbitrum bridge").setAction(
+  async function (_, hre_) {
+    const hre = hre_ as CombinedHRE;
+    const { deployments, getNamedAccounts, web3 } = hre;
+    const { utf8ToHex } = web3.utils;
+    const { deployer } = await getNamedAccounts();
+
+    const Finder = await deployments.get("Finder");
+    const finder = new web3.eth.Contract(Finder.abi, Finder.address);
+    const ChildMessenger = await deployments.get("Arbitrum_ChildMessenger");
```

Comment:
Is the name of this contract the only thing that would differ between this setup and the setup for other L2s?

In hardhat, you can optionally _name_ a deployment something other than its contract name, so if we wanted to generalize, we could potentially utilize that feature.

Reply:
> you can optionally name a deployment something other than its contract name,

Oh? that's pretty interesting. I think the child messenger set up wil be the same but I also think its just as clean to run something like `_setupChildMessenger(nameOfChildMessengerContract)`

Reply:
Yep, that works too! No strong preference here